openstack/resource_openstack_compute_instance_v2.go

```
@@ -599,6 +648,57 @@ func resourceComputeInstanceV2Update(d *schema.ResourceData, meta interface{}) e
 		}
 	}
 
+	if d.Get("state").(string) != "" {
+		if d.HasChange("state") {
+			vmState := d.Get("state").(string)
+			if strings.ToLower(vmState) == "shutoff" {
+				err = startstop.Stop(computeClient, d.Id()).ExtractErr()
+				if err != nil {
+					log.Printf("[WARN] Error stopping OpenStack instance: %s", err)
```

We should return an error rather than log one.